packages/sign_in_with_apple/lib/src/authorization_credential.dart

```diff
@@ -22,10 +22,10 @@ class AuthorizationCredentialAppleID {
     @required this.givenName,
     @required this.familyName,
     @required this.email,
-    @required this.authorizationCode,
+    @required String? authorizationCode,
     @required this.identityToken,
     @required this.state,
-  }) {
+  }) : authorizationCode = authorizationCode ?? '' {
```

Comment:
This check we might want to move before the constructor, as it's a little unwieldy here.

Reply:
An authorization code is also always expected, right?

Reply:
Wait, expanding the code below answers the question 😁 

I guess this should move to the caller then as I'm also not sure to which `authorizationCode` the constructor actually refers to here (deducting from NNDB that it is the value from the constructor and not the class)

Reply:
Moved the check to the  caller (before calling the constructor), where we can also better log the source of the missing data.

Changed in d45665d0504467c7c372a01c9a8ae66c475b7ad3